### PR TITLE
[TF FE][Opset13] Enable Multinomial operator in TF frontend

### PR DESCRIPTION
**Pull request by mmikolajcz**
### Details:
 - *Add translator for multinomial operator*
 - *Add test-cases*

### Tickets:
 - *118986*

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        tf.compat.v1.reset_default_graph()
+        # Configuration required to make multinomial randomness predictable across devices, results depends on TF parallel execution.
+        session_conf = tf.compat.v1.ConfigProto(intra_op_parallelism_threads=1, inter_op_parallelism_threads=1)
+        # Create the graph and model
+        with tf.compat.v1.Session(config=session_conf) as sess:
+            probs = tf.compat.v1.placeholder(input_type, logits_shape, "probs")
+            num_samples = tf.compat.v1.placeholder(tf.int32, [], "num_samples")
+            if global_seed is not None:
+                tf.random.set_seed(global_seed)
+            tf.random.categorical(tf.math.log(probs), num_samples, dtype=out_type, seed=op_seed)
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py:26]: please use https://www.tensorflow.org/api_docs/python/tf/raw_ops/Multinomial in the test because we don't know how `categorical` is decomposed and if it contains needed operation

File: tests/layer_tests/pytest.ini [hunk 2]
```diff
@@ -2,6 +2,7 @@
 markers =
     nightly
     precommit
+    precommit_tf_fe
```
Comment (rkazants) [inline tests/layer_tests/pytest.ini:5]:
```suggestion
```

Comment (rkazants) [reply tests/layer_tests/pytest.ini:5]: Not needed, all setup is correct already

File: src/frontends/tensorflow_common/src/op/multinomial.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "common_op_table.hpp"
+#include "openvino/opsets/opset13.hpp"
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/multinomial.cpp:6]: please use explicit include of Multinomial operation. We avoid common include usage now.

File: tests/layer_tests/pytest.ini (hunk 2 above)
Comment (mmikolajcz) [reply tests/layer_tests/pytest.ini:5]: It is not needed for pytest to work, however, without it, around ~130 warnings to add it are raised, possibly masking important warnings and creating around 500l to scroll to output. It shouldn't break or fix anything, just make logs more clear with just 2 short warnings left

File: tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py (hunk 1 above)
Comment (mmikolajcz) [reply tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py]: Done

File: src/frontends/tensorflow_common/src/op/multinomial.cpp (hunk 3 above)
Comment (mmikolajcz) [reply src/frontends/tensorflow_common/src/op/multinomial.cpp]: Done

**Comment by rkazants** [comment]
@mmikolajcz, @mlukasze, what are our plans for this operation? Is it moved to 2023.3 release, right?

File: tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            ),
+            (
+                np.array([[7, 7, 7, 7], [7, 7, 7, 7], [7, 7, 7, 7]], dtype=np.float32),
+                1024,
+                [32465, None],
+            ),
+        ],
+    )
+    @pytest.mark.nightly
+    @pytest.mark.precommit
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py:65]: `precommit` is not needed since it targets the legacy FE that is not needed
```suggestion
```

**Review by gkrivor — APPROVED** [review]

File: tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+                ),
+                256,
+                [32465, 48971],
+                "shape",
+            ),
+            (np.array([[1, 1, 1, 1]], dtype=np.float16), 1024, [1, 1], "shape"),
+            (
+                np.array([[1, 2, 3, 4], [4, 3, 2, 1], [1, 0, 0, 0]], dtype=np.float32),
+                1,
+                [78132, None],
```
Comment (PiotrKrzem) [inline tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py:99]:
```suggestion
                [78132, 0],
```

File: tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+                "shape",
+            ),
+            (
+                np.array([[7, 7, 7, 7], [7, 7, 7, 7], [7, 7, 7, 7]], dtype=np.float32),
+                1024,
+                [32465, None],
```
Comment (PiotrKrzem) [inline tests/layer_tests/tensorflow_tests/test_tf_Multinomial.py:105]:
```suggestion
                [32465, 0],
```

**Review by PiotrKrzem — APPROVED** [review]
Suggestion: stylistic change from seed = None to seed = 0, otherwise LGTM!
